File: .tmp_py4013/batch_04_sources/2455_average_value_of_even_numbers_that_are_divisible_by_three.js
// LeetCode 2455 - Average Value of Even Numbers That Are Divisible by Three
// https://leetcode.com/problems/average-value-of-even-numbers-that-are-divisible-by-three/

/**
 * @param {number[]} nums
 * @return {number}
 */
var averageValue = function(nums) {
    let sum = 0, cnt = 0;
    for (const x of nums) {
        if (x % 6 === 0) {
            sum += x;
            cnt++;
        }
    }
    return cnt === 0 ? 0 : Math.floor(sum / cnt);
};

File: .tmp_rs4013/rust_check_batch08/2455_average_value_of_even_numbers_that_are_divisible_by_three.rs
struct Solution;
// LeetCode 2455 - Average Value of Even Numbers That Are Divisible by Three
// https://leetcode.com/problems/average-value-of-even-numbers-that-are-divisible-by-three/

impl Solution {
    pub fn average_value(nums: Vec<i32>) -> i32 {
        let mut sum = 0;
        let mut cnt = 0;
        for x in nums {
            if x % 6 == 0 {
                sum += x;
                cnt += 1;
            }
        }
        if cnt == 0 {
            0
        } else {
            sum / cnt
        }
    }
}

fn main() {}
